annotation-utils/src/main/java/eu/europeana/annotation/utils/parse/AnnotationLdParser.java

```diff
@@ -592,6 +590,25 @@ private Body parseBody(JSONObject valueObject, MotivationTypes motivation) throw
 				case WebAnnotationFields.LONGITUDE:
 					setLongitude(body, value.toString());
 					break;
+					// parsing for semantic tag Vcard address
+				case WebAnnotationFields.STREET_ADDRESS:
```

Comment:
Please move the parsing of address properties to own method, which must be invoked only if the body type is VcardAddress, the same refactoring must be performed for Agent body

Reply:
extracted own methods for VcardAddress and for Place. I assume you do mean Place when you write about Agent body.

Reply:
maybe I was looking in the wrong branch, but we have also the properties for AgendBody that need to be parsed i nthe same way

Reply:
Fixed Agent fields